process/scToProtocol/stakingToPeer_test.go

```diff
@@ -0,0 +1,661 @@
+	"github.com/ElrondNetwork/elrond-go/vm/factory"
+	"github.com/ElrondNetwork/elrond-go/vm/systemSmartContracts"
+	vmcommon "github.com/ElrondNetwork/elrond-vm-common"
+	"github.com/stretchr/testify/assert"
+)
+
+func createMockArgumentsNewStakingToPeer() ArgStakingToPeer {
+	return ArgStakingToPeer{
+		AdrConv:      &mock.AddressConverterMock{},
+		Hasher:       &mock.HasherMock{},
+		Marshalizer:  &mock.MarshalizerStub{},
+		PeerState:    &mock.AccountsStub{},
+		BaseState:    &mock.AccountsStub{},
+		ArgParser:    &mock.ArgumentParserMock{},
+		CurrTxs:      &mock.TxForCurrentBlockStub{},
+		ScDataGetter: &mock.ScDataGetterMock{},
+	}
+}
+
+func createBlockBody() block.Body {
+	return block.Body{
+		{
+			TxHashes:        [][]byte{[]byte("hash1"), []byte("hash2")},
+			ReceiverShardID: 0,
+			SenderShardID:   0,
+			Type:            block.SmartContractResultBlock,
+		},
+	}
+}
+
+func TestNewStakingToPeerNilAddrConverterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.AdrConv = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilAddressConverter, err)
+}
+
+func TestNewStakingToPeerNilHasherShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.Hasher = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilHasher, err)
+}
+
+func TestNewStakingToPeerNilMarshalizerShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.Marshalizer = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilMarshalizer, err)
+}
+
+func TestNewStakingToPeerNilPeerAccountAdapterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.PeerState = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilPeerAccountsAdapter, err)
+}
+
+func TestNewStakingToPeerNilBaseAccountAdapterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.BaseState = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilAccountsAdapter, err)
+}
+
+func TestNewStakingToPeerNilArgumentParserShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.ArgParser = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilArgumentParser, err)
+}
+
+func TestNewStakingToPeerNilCurrentBlockHeaderShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.CurrTxs = nil
+
+	_, err := NewStakingToPeer(arguments)
+	assert.Equal(t, process.ErrNilTxForCurrentBlockHandler, err)
+}
+
+func TestNewStakingToPeerNilScDataGetterShouldErr(t *testing.T) {
+	t.Parallel()
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.ScDataGetter = nil
+
+	_, err := NewStakingToPeer(arguments)
```

Comment:
you can also check that the resulted `stakingToPeer` is nil in this test and the tests above it

Reply:
Done.